sklearn/utils/tests/test_sparsefuncs.py

```diff
@@ -156,6 +164,32 @@ def test_incr_mean_variance_axis():
                 assert_array_equal(X.shape[axis], n_incr)
 
 
+def test_incr_mean_variance_axis_dim_mismatch():
+    n_samples, n_features = 60, 4
+    rng = np.random.RandomState(42)
+    X = sp.csc_matrix(rng.rand(n_samples, n_features))
+
+    last_mean = np.zeros(n_features)
+    last_var = np.zeros_like(last_mean)
+    last_n = np.zeros(last_mean.shape, dtype=np.int64)
+
+    kwargs = dict(last_mean=last_mean, last_var=last_var, last_n=last_n)
+    mean0, var0, _ = incr_mean_variance_axis(X, axis=0, **kwargs)
+    assert_array_almost_equal(np.mean(X.toarray(), axis=0), mean0)
+    assert_array_almost_equal(np.var(X.toarray(), axis=0), var0)
+
+    # test ValueError if axis=1 and last_mean.size == n_features
+    with pytest.raises(ValueError):
+        mean1, var1, _ = incr_mean_variance_axis(X, axis=1, **kwargs)
+        assert_array_almost_equal(np.mean(X.toarray(), axis=1), mean1)
+        assert_array_almost_equal(np.var(X.toarray(), axis=1), var1)
```

Comment:
Why call `assert_array_almost_equal`  after `incr_mean_variance_axis` is expected to fail?

```suggestion
    with pytest.raises(ValueError):
        incr_mean_variance_axis(X, axis=1, **kwargs)
```